### PR TITLE
Public Key authentication

### PR DESCRIPTION
**Pull request by amao9098**
* Modify to accept Password Authentication or Public Key authentication
* change buffer length for tcp

### CODE REVIEW

File: runPerftool.psm1 [hunk 1]
```diff
@@ -332,55 +358,58 @@ Function ProcessToolCommands{
...
         # delay to let credential (public key) propagate before remoting
-        $credPropagationTimeInSecond = 2
+        $credPropagationTimeInSecond = 3
     
         [System.IO.TextReader] $recvCommands = $null
         [System.IO.TextReader] $sendCommands = $null
     
         $toolpath = "./{0}" -f $Toolname
-    
         $homePath = "/home/$TestUserName"
```
Comment (cmaloney-msft) [inline runPerftool.psm1:367]: Should we just use the ~ shortcut here? Do we always assume the passed in user name is the same as the running user?

File: runPerftool.psm1 [hunk 2]
```diff
@@ -209,14 +218,19 @@ Function ProcessCommands{
...
+    [Parameter(Mandatory=$False, Position=0, HelpMessage="Dest Machine Key File?")]
+    [String]$DestIpKeyFile = "",
     [Parameter(Mandatory=$True, Position=0, HelpMessage="Src Machine Username?")]
     [string] $SrcIpUserName,
     [Parameter(Mandatory=$True, Position=0, HelpMessage="Src Machine Password?")]
     [SecureString]$SrcIpPassword,
+    [Parameter(Mandatory=$False, Position=0, HelpMessage="Src Machine Key File?")]
+    [String]$SrcIpKeyFile = "",
     [Parameter(Mandatory=$True, Position=0, HelpMessage="Test Machine Username?")]
     [string] $TestUserName,
```
Comment (cmaloney-msft) [inline runPerftool.psm1:235]: Is this only used for building the path?

File: runPerftool.psm1 (hunk 1 above)
Comment (amao9098) [reply runPerftool.psm1:367]: When running on elevated privileges with Invoke-AzRunCommand, ~ does not exist.

File: runPerftool.psm1 (hunk 2 above)
Comment (amao9098) [reply runPerftool.psm1:235]: Yes, for the path to the private key that will give access to the remote VMs.

File: SetupTearDown.ps1 [hunk 3]
```diff
@@ -14,7 +14,11 @@
 
 .PARAMETER Password
     Required Parameter. Get the password of this computer to modify firewall permissions.
-.DESCRIPTION
+
+.PARAMETER PassAuth
+    Required Parameter. Get the password of this computer to modify firewall permissions.
```
Comment (vidyas00) [inline SetupTearDown.ps1:19]: optional

**Review by vidyas00 — APPROVED** [review]

**Pull request merged**